src/Core/Services/NoopImplementations/NoopAttachmentStorageService.cs

```diff
@@ -32,24 +33,30 @@ public Task DeleteAttachmentsForUserAsync(Guid userId)
             return Task.FromResult(0);
         }
 
-        public Task RollbackShareAttachmentAsync(Guid cipherId, Guid organizationId, string attachmentId)
+        public Task RollbackShareAttachmentAsync(Guid cipherId, Guid organizationId, CipherAttachment.MetaData attachmentData, string originalContainer)
         {
             return Task.FromResult(0);
         }
 
-        public Task StartShareAttachmentAsync(Guid cipherId, Guid organizationId, string attachmentId)
+        public Task StartShareAttachmentAsync(Guid cipherId, Guid organizationId, CipherAttachment.MetaData attachmentData)
         {
             return Task.FromResult(0);
         }
 
-        public Task UploadNewAttachmentAsync(Stream stream, Cipher cipher, string attachmentId)
+        public Task UploadNewAttachmentAsync(Stream stream, Cipher cipher, CipherAttachment.MetaData attachmentData)
         {
             return Task.FromResult(0);
         }
 
-        public Task UploadShareAttachmentAsync(Stream stream, Guid cipherId, Guid organizationId, string attachmentId)
+        public Task UploadShareAttachmentAsync(Stream stream, Guid cipherId, Guid organizationId, CipherAttachment.MetaData attachmentData)
         {
             return Task.FromResult(0);
         }
+
+        public Task<string> GetAttachmentDownloadUrlAsync(Cipher cipher, CipherAttachment.MetaData attachmentData)
+        {
+            return Task.FromResult(string.Empty);
```

Comment:
why not null? Would that break upstream?

Reply:
As long as the clients handled the case it'd be OK. That URL is shipped directly off to the client requesting the URL.

I'm happy to change, I don't have any preference for one over the other

Reply:
Yeah, let's return `null`

Reply:
Changed to null